lib/src/ctx/cancellation.rs

```diff
@@ -0,0 +1,24 @@
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use std::time::Instant;
+
+/// A 'static view into the cancellation status of a Context.
+#[derive(Clone, Debug, Default)]
+pub struct Cancellation {
+	deadline: Option<Instant>,
+	cancellations: Vec<Arc<AtomicBool>>,
+}
```

Comment:
I did plenty of experimentation before settling on this design, which leaves `Context` pretty much as-is (non-`'static`). As long as `Context` is non-`'static`, then it can't be shared between threads or used by rQuickJS's interrupt feature.